adafruit_esp32spi/adafruit_esp32spi.py

```diff
@@ -131,6 +131,97 @@
 # pylint: disable=too-many-lines
 
 
+class ESP_Network:
+    """A wifi network provided by a nearby access point."""
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        esp_spi_control=None,
+        raw_ssid=None,
+        raw_bssid=None,
+        raw_rssi=None,
+        raw_channel=None,
+        raw_country=None,
+        raw_authmode=None,
+    ):
+        self._esp_spi_control = esp_spi_control
+        self._raw_ssid = raw_ssid
+        self._raw_bssid = raw_bssid
+        self._raw_rssi = raw_rssi
+        self._raw_channel = raw_channel
+        self._raw_country = raw_country
+        self._raw_authmode = raw_authmode
+
+    def _get_response(self, cmd):
+        respose = self._esp_spi_control._send_command_get_response(  # pylint: disable=protected-access
+            cmd, [b"\xFF"]
+        )
+        return respose[0]
+
+    @property
+    def ssid(self):
+        """String id of the network"""
+        if self._raw_ssid:
+            response = self._raw_ssid
+        else:
+            response = self._get_response(_GET_CURR_SSID_CMD)
+        return response.decode("utf-8")
+
+    @property
+    def bssid(self):
+        """BSSID of the network (usually the AP’s MAC address)"""
+        if self._raw_bssid:
+            response = self._raw_bssid
+        else:
+            response = self._get_response(_GET_CURR_BSSID_CMD)
+        return bytes(response)
+
+    @property
+    def rssi(self):
+        """Signal strength of the network"""
+        if self._raw_bssid:
+            response = self._raw_rssi
+        else:
+            response = self._get_response(_GET_CURR_RSSI_CMD)
+        return struct.unpack("<i", response)[0]
+
+    @property
+    def channel(self):
+        """Channel number the network is operating on"""
+        if self._raw_channel:
+            return self._raw_channel[0]
+        return None
+
+    @property
+    def country(self):
+        """String id of the country code"""
+        if self._raw_country:
+            return self._raw_country
+        return None
```

Comment:
If self._raw_country is false-y, then it's None, so the `if` is unnecessary, right?
```suggestion
        return self._raw_country
```